staging/src/k8s.io/apimachinery/pkg/apimachinery/announced/group_factory.go

```diff
@@ -89,6 +89,17 @@ func (gmf *GroupMetaFactory) Announce(groupFactoryRegistry APIGroupFactoryRegist
 		panic(err)
 	}
 	return gmf
+
+}
+
+// AnnounceIfNotAnnounced is the same as Announce, except that it returns nil
+// if gmf.GroupArgs.GroupName is already registered in groupFactoryRegistry,
+// instead of panicing.
+func (gmf *GroupMetaFactory) AnnounceIfNotAnnounced(groupFactoryRegistry APIGroupFactoryRegistry) *GroupMetaFactory {
+	if _, ok := groupFactoryRegistry[gmf.GroupArgs.GroupName]; ok {
```

Comment:
one could put some concistency checks here, i.e. if you re-announce with different values. Just an idea...